custom_components/elevenlabs_tts/config_flow.py

```diff
@@ -0,0 +1,120 @@
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+import voluptuous as vol
+
+from .const import (
+    CONF_MODEL,
+    CONF_OPTIMIZE_LATENCY,
+    CONF_SIMILARITY,
+    CONF_STABILITY,
+    CONF_VOICE,
+    DEFAULT_MODEL,
+    DEFAULT_OPTIMIZE_LATENCY,
+    DEFAULT_SIMILARITY,
+    DEFAULT_STABILITY,
+    DEFAULT_VOICE,
+    DOMAIN,
+)
+from .elevenlabs import ElevenLabsClient
+
+
+class ElevenlabsTTSSetupFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    VERSION = 1
+
+    async def async_step_import(self, import_config):
+        """Import a config entry from configuration.yaml."""
+        return await self.async_step_user(import_config)
+
+    async def async_step_user(self, user_input: dict = None) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            # Validate the provided API key
+            if not await self._validate_api_key(user_input[CONF_API_KEY]):
+                errors[CONF_API_KEY] = "invalid_api_key"
+
+            if not errors:
+                return self.async_create_entry(title="Eleven Labs TTS", data=user_input)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_API_KEY): str,
+                }
+            ),
+            errors=errors,
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return ElevenlabsTTSOptionsFlowHandler(config_entry)
+
+    async def _validate_api_key(self, api_key):
+        """Perform API key validation."""
+        # Implement your validation logic here, e.g., make an API
+        # call to validate the key
+        # Return True if the key is valid, otherwise False
+        client = ElevenLabsClient(self.hass, api_key=api_key)
+        await client.get_voices()
+        return True
```

Comment:
Putting in a bogus key to test the sad path caused the integration to accept the bad key, finish setup, and log errors.

This will fix key validation
```suggestion
        voices = await client.get_voices()
        return bool(voices)
```